mne/preprocessing/detect_bad_channels.py

```diff
@@ -0,0 +1,67 @@
+# Authors: Velu Prabhakar Kumaravel <vkumaravel@fbk.eu>      
+# License: BSD-3-Clause
+
+
+import numpy as np
+
+from ..io.base import BaseRaw
+from .. import Transform
+from ..utils import (logger, verbose, _validate_type)
+
+
+@verbose
+def mark_bad_channels_lof(raw, n_neighbors = 8, metric = 'sqeuclidean', lof_threshold = 1.5):
+    """ Detects and marks bad channels using Local Outlier Factor (LOF) algorithm.
+
+    Detects bad (a.k.a. anamolous) channels using a local density based algorithm:LOF.
+    By default, n_neighbors parameter is set to 8, which will work in most cases
+
+    See :footcite:`kumaravel2022` for background on choosing
+    ``metric`` and ``lof_threshold``.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data.
+
+    n_neighbors : int
+        Number of neighbors for computing local density (default: 8).
+
+    metric : 'euclidean' | 'sqeuclidean' | 'cityblock' | 'cosine' | 'l1' | 'l2' | 'manhattan' | 'nan_euclidean'
+        The metric used for distance compuation. Default: 'sqeuclidean' as it performed better than 'euclidean' in MATLAB implementation.
+        One could consider this as an hyperparameter to optimize.
+
+    lof_threshold : float
```

Comment:
probably just "threshold" is a better parameter name choice here.  Also would be good to say what the allowable / sensible range is...  the notes already make clear that it is not restricted to `0 < threshold <= 1`, but are negative values meaningful here?  What is a sensible upper bound on threshold? etc.